src/components/ConfirmationButton/ConfirmationButton.stories.mdx

```diff
@@ -0,0 +1,127 @@
+import { ArgsTable, Canvas, Meta, Story } from "@storybook/addon-docs";
+import { useState } from "react";
+
+import ConfirmationButton from "./ConfirmationButton";
+import Input from "../Input";
+
+<Meta title="ConfirmationButton" component={ConfirmationButton} />
+
+### ConfirmationButton
+
+`ConfirmationButton` is a specialised version of the Vanilla [Button](https://docs.vanillaframework.io/patterns/buttons/) that prompts a confirmation from the user before executing some action.
+
+### Props
+
+<ArgsTable of={ConfirmationButton} />
+
+export const doNothing = () => {};
+
+### Default
+
+<Canvas>
+  <Story name="Default">
+    <ConfirmationButton
+      title="Confirm delete"
+      buttonLabel="Delete"
+      confirmButtonLabel="Delete"
+      onConfirm={doNothing}
+    >
+      <p>
+        Are you sure you want to delete user "Simon"?<br/>
+        This action is permanent and can not be undone.
+      </p>
+    </ConfirmationButton>
+  </Story>
+</Canvas>
+
+### Icon and label
+
+<Canvas>
+  <Story name="Icon and label">
+    {() => {
+      const [isLoading, setLoading] = useState(false);
+      const brieflyLoad = () => {
+        setLoading(true);
+        setTimeout(() => setLoading(false), 2000);
+      };
+      return (
+        <ConfirmationButton
+          title="Confirm delete"
+          buttonLabel="Delete"
+          confirmButtonLabel="Delete"
+          icon="delete"
+          isLoading={isLoading}
+          onConfirm={brieflyLoad}
+          shiftClickEnabled
+          showShiftClickHint
+        >
+          <p>
+            Are you sure you want to delete user "Simon"?<br/>
+            This action is permanent and can not be undone.
+          </p>
+        </ConfirmationButton>
+      );
+    }}
+  </Story>
+</Canvas>
+
+### Icon only
+
+<Canvas>
+  <Story name="Icon only">
+    {() => {
+      const [isLoading, setLoading] = useState(false);
+      const brieflyLoad = () => {
+        setLoading(true);
+        setTimeout(() => setLoading(false), 2000);
+      };
+      return (
+        <ConfirmationButton
+          title="Confirm delete"
+          confirmButtonLabel="Delete"
+          icon="delete"
+          isLoading={isLoading}
+          onConfirm={brieflyLoad}
+          shiftClickEnabled
+          showShiftClickHint
+        >
+          <p>
+            Are you sure you want to delete user "Simon"?<br/>
+            This action is permanent and can not be undone.
+          </p>
+        </ConfirmationButton>
+      );
+    }}
+  </Story>
+</Canvas>
+
+### Base appearance
+
+<Canvas>
+  <Story name="Base appearance">
+    {() => {
+      const [isLoading, setLoading] = useState(false);
+      const brieflyLoad = () => {
+        setLoading(true);
+        setTimeout(() => setLoading(false), 2000);
```

Comment:
For some reason these states aren't doing anything for me.

Reply:
Really? Strange, this is what I see when I click on the buttons with the loading state:

![Peek 2023-06-16 10-06](https://github.com/canonical/react-components/assets/56583786/010e422a-c1b2-40e8-8c03-c7e02bac9f78)

This state was added just to provide a quick example of a fake "loading" operation starting upon confirmation.

Reply:
Oh, ignore me, I thought the spinner was supposed to appear when you click the button to open the modal, but it actually appears when you confirm!